File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultCachedVoice.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultCachedVoice extends InlineQueryResult<InlineQueryResultCachedVoice> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String voice_file_id;
    private String title;

    private String caption;
    private String parse_mode;

    public InlineQueryResultCachedVoice(String id, String fileId, String title) {
        super("voice", id);
        this.voice_file_id = fileId;
        this.title = title;
    }

    public InlineQueryResultCachedVoice caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultCachedVoice parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultCachedVoice.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultCachedVoice(id: String, private val voice_file_id: String, private val title: String) : InlineQueryResult<InlineQueryResultCachedVoice>("voice", id), Serializable {

    private var caption: String? = null
    private var parse_mode: String? = null

    fun caption(caption: String): InlineQueryResultCachedVoice {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultCachedVoice {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
